crates/driver/src/infra/config/file/mod.rs

```diff
@@ -265,6 +266,10 @@ struct SolverConfig {
     /// Which `tx.origin` is required to make a quote simulation pass.
     #[serde(default)]
     quote_tx_origin: Option<eth::H160>,
+
+    /// List of CoW AMM addresses
+    #[serde(default)]
+    cow_amm_addresses: Vec<H160>,
```

Comment:
Can we add this list to the auction that gets sent to the driver (I believe the autopilot already has this list). This way co-located drivers could also settle CoW AMMs.

Reply:
I'm a bit torn on this. CoW AMMs are nothing inherent to the protocol so putting them into the data sent to the driver seems odd to me. Also CoW AMMs could be understood as a liquidity source which gets managed by the driver. This might be a bit misguided though because once CoW AMM orders can contribute to the solution score with their surplus it could be perfectly reasonable for a solver to completely ignore user orders and only settle CoW AMM orders (meaning they would not be used to provide liquidity for other orders).

Reply:
I wouldn't put "CoW AMMs" in the file, but instead call them "surplus capturing jit-order owners" or the like. It's true that eventually we will need a formal definition of what qualifies as such orders (I assume we will allow-list certain smart order factories for which we know they create "good" contract instances) but whatever the logic is the autopilot needs to implement it to double check the score computation.

For now it's a hardcoded list and so I think it should be sent along with the auction on every batch. How else would we communicate to solvers which addresses are eligible?

Reply:
> For now it's a hardcoded list and so I think it should be sent along with the auction on every batch. How else would we communicate to solvers which addresses are eligible?

I think for now it's a reasonable suggestion to send the list with the request but eventually when we have possibly many CoW AMMs I would just expect the driver to compute the score based on what it thinks the available AMMs are and let the autopilot do the same.

Reply:
Guys @fleupold @MartinquaXD I think I reworked it as suggested! please let me know! :pray: